src/sqlfluff/rules/L064.py

```diff
@@ -0,0 +1,234 @@
+    },
+}
+# BigQuery string prefix characters.
+STRING_PREFIX_CHARS: Final = "rbRB"
+
+
+@document_configuration
+@document_fix_compatible
+class Rule_L064(BaseRule):
+    r"""Consistent usage of preferred quotes for STRING datatype.
+
+    Some databases allow single quotes as well as double quotes to express a STRING
+    literal. Prefer one type of quotes as specified in rule setting. Falling back to
+    alternate quotes to reduce escapes.
+
+    Dollar quotes raw strings are excluded from this rule, as they mostly used for
+    literal UDF Body definitions.
+
+    **Anti-pattern**
+
+    .. code-block:: sql
+       :force:
+
+        select
+            "abc",
+            'abc',
+            "\"",
+            "abc" = 'abc'
+        from foo
+
+    **Best practice**
+
+    Ensure all STRING literals use preferred quotes, unless escaping can be reduced
+    using alternate quotes.
+
+    .. code-block:: sql
+       :force:
+
+        select
+            "abc",
+            "abc",
+            '"',
+            "abc" = "abc"
+        from foo
+
+    """
+
+    config_keywords = ["preferred_string_quotes", "force_enable"]
+    _dialects_with_double_quoted_strings = [
+        "bigquery",
+        "hive",
+        "mysql",
+        "sparksql",
+    ]
+
+    def _eval(self, context: RuleContext) -> Optional[LintResult]:
+        # Config type hints
+        self.preferred_string_quotes: str
+        self.force_enable: bool
+
+        if (
+            context.dialect.name not in self._dialects_with_double_quoted_strings
+            and not self.force_enable
+        ):
+            return LintResult()
+
+        # Only care about quoted literal segments.
+        if not context.segment.name == "quoted_literal":
+            return None
+
+        fixed_string = self._normalize_preferred_string_quotes(
+            context.segment.raw,
+            preferred_quote_char=QUOTES_MAPPING[self.preferred_string_quotes][
+                "preferred_quote_char"
+            ],
+            alternate_quote_char=QUOTES_MAPPING[self.preferred_string_quotes][
+                "alternate_quote_char"
+            ],
+        )
+
+        if fixed_string != context.segment.raw:
+            return LintResult(
+                anchor=context.segment,
+                fixes=[
+                    LintFix.replace(
+                        context.segment,
+                        [
+                            CodeSegment(
+                                raw=fixed_string,
+                                name="quoted_literal",
+                                type="literal",
+                            )
+                        ],
+                    )
+                ],
+                description=(
+                    "Inconsistent use of preferred quote style '"
+                    f"{QUOTES_MAPPING[self.preferred_string_quotes]['common_name']}'. "
+                    f"Use {fixed_string} instead of {context.segment.raw}."
```

Comment:
Can we omit including the actual string in the lint message? Reasons:
* The string could be quite long.
* In some cases, the string may contain sensitive information that shouldn't appear in a build log.

Reply:
Quite a few other rules use this, and it can be handy.

Not sure I totally agree with the security concerns since this is raw text from SQL file.

Reply:
In scenarios where the string contains a lot of quotes it might be very hard for the user to tell what is the "correct" way of quoting. So, when users only use the linter (for example as a plugin to their editor) I imagined it to be rather frustrating if we only tell them that there is a problem but not what the solution is.

I already removed this, let me know if I can revert this.

Reply:
I don't feel strongly about this, so it's good to add back. 👍